lib/ReportHistoryStore.jsx

```diff
@@ -191,4 +299,29 @@ export default class ReportHistoryStore {
 
         return promise.resolve(cachedHistory);
     }
+
+    /**
+     * Gets the history from the cache if it exists. Otherwise fully loads the history.
+     *
+     * @param {Number} reportID
+     *
+     * @return {Deferrred}
+     */
+    getFromCacheByActionID(reportID) {
+        const promise = new Deferred();
+        const cachedHistory = this.cache[reportID] || [];
+
+        // First check to see if we even have this history in cache
+        if (_.isEmpty(cachedHistory)) {
+            this.API.Report_GetHistory({reportID})
+                .done((reportHistory) => {
+                    this.mergeHistoryByTimestamp(reportID, reportHistory);
+                    promise.resolve(this.cache[reportID]);
+                })
+                .fail(promise.reject);
+            return promise;
```

Comment:
frankly i copied over few methods like get, mergeItems, insertIntoCache and getFromCache and tweaked it just a little bit so that it would support deprecating sequence numbers. But yes here we can call `getByActionID` directly. I'll update it.